configure

```diff
@@ -234,11 +240,16 @@ fi
 
 # directories
 
-if test "$ocamllibdir" = "auto"; then ocamllibdir=`ocamlc -where`; fi
-
-# fails on Cygwin:
-# if test ! -f "$ocamllibdir/caml/mlvalues.h"
-# then echo "cannot find OCaml libraries in $ocamllibdir"; exit 2; fi
+if test "$ocamllibdir" = "auto"; then
+  ocamllibdir=`ocamlc -where`;
+  if test "$OSTYPE" = "cygwin"; then
+    # on Windows, ocamlc -where outputs the path to OCaml's library
+    # directory, followed by a carriage return. The latter needs to
+    # be removed and a trailing slash added in order to obtain a
+    # Unix-style path that is accepted by gcc on Cygwin
+    ocamllibdir="`cygpath -u $ocamllibdir | tr -d '\r'`/";
```

Comment:
The comment doesn't quite match the code, as the call to `cygpath -u` is not explained.  But that's a minor nit.  

A less minor concern is that this test assumes that if the configure script is run under Cygwin, then the native Win32 port of OCaml is used, as opposed to the Cygwin port.  Both are actually possible.  The problem existed before this pull request, when selecting `arch='x86_64_mingw64'` below.  What this means is that Zarith should also support the Cygwin port of OCaml and doesn't, currently.